common/membership/ringpop/factory.go

```diff
@@ -144,13 +145,25 @@ func (factory *factory) getMonitor() *monitor {
 			factory.broadcastAddressResolver,
 			factory.Config.MaxJoinDuration,
 			maxPropagationTime,
-			time.Time{}, // TODO: set this based on dynamic config
+			factory.getJoinTime(maxPropagationTime),
 		)
 	})
 
 	return factory.monitor
 }
 
+func (factory *factory) getJoinTime(maxPropagationTime time.Duration) time.Time {
+	var alignTime time.Duration
+	switch factory.ServiceName {
```

Comment:
Would love it if you disabled `unnecessary-stmt` in the revive settings of our linter. This rule seems super unnecessary.